src/taskCommands.js

```
@@ -151,4 +152,176 @@ export function createTask(projectRoot, options) {
   writeFileSync(filePath, content, "utf8");
 
   return { taskId, filePath };
+}
+
+/**
+ * Parses a task file and extracts metadata from YAML frontmatter
+ */
+function parseTaskFile(filePath) {
+  try {
+    const content = readFileSync(filePath, "utf8");
+    const filename = basename(filePath, extname(filePath));
+
+    // Extract task ID from filename
+    const taskId = filename;
+
+    // Parse YAML frontmatter
+    const frontmatterMatch = content.match(/^---\n([\s\S]*?)\n---/);
+    if (!frontmatterMatch) {
+      console.error(`Warning: No frontmatter found in ${filePath}`);
+      return null;
+    }
+
+    const frontmatter = frontmatterMatch[1];
+    const metadata = YAML.load(frontmatter);
+
+    return {
+      taskId,
+      title: metadata.title || "Untitled",
+      status: metadata.status || "todo",
+      priority: metadata.priority || "low"
+    };
+  } catch (error) {
+    console.error(`Error parsing task file ${filePath}:`, error.message);
```

throw hard error. no silent errors

can remove entire try catch